src/main/java/org/gridsuite/filter/utils/expertfilter/ExpertFilterUtils.java

```diff
@@ -204,25 +215,28 @@ private static String getSectionBasedFieldValue(FieldType field, ShuntCompensato
         };
     }
 
-    private static String getBatteryFieldValue(FieldType field, Battery battery) {
+    private static String getBatteryFieldValue(FieldType field, String propertyName, Battery battery) {
         return switch (field) {
             case COUNTRY,
                     NOMINAL_VOLTAGE,
-                    VOLTAGE_LEVEL_ID -> getVoltageLevelFieldValue(field, battery.getTerminal().getVoltageLevel());
+                    VOLTAGE_LEVEL_ID -> getVoltageLevelFieldValue(field, null, battery.getTerminal().getVoltageLevel());
             case CONNECTED -> getTerminalFieldValue(field, battery.getTerminal());
             case MIN_P -> String.valueOf(battery.getMinP());
             case MAX_P -> String.valueOf(battery.getMaxP());
             case TARGET_P -> String.valueOf(battery.getTargetP());
             case TARGET_Q -> String.valueOf(battery.getTargetQ());
+            case FREE_PROPERTIES -> battery.getProperty(propertyName);
```

Comment:
not needed as we add a case in getFieldValue

Reply:
done